src/model/user/user.schema.ts

```diff
@@ -0,0 +1,16 @@
+import { Schema, model } from "mongoose";
+import { IUser } from "./user";
+
+const userSchema = new Schema<IUser>({
+
+    firstName: { type: String, required: true },
+    lastName: { type: String, required: true },
+    email: {type: String, required: true},
+    hashedPassword: {type: String, required: true},
+    bio: {type: String, required: true},
+    recipes: [{ key: String, value: Date}]

```

Comment:
So for this one, we could store recipes in an array. I was thinking that it might be more efficient to store recipes as an object (key value pairs, where the key is the recipe _id and the value is maybe a date object).

Example:
recipes : {
   recipeId001: <timeStamp>,
   recipeId002: <timeStamp>,
   recipeId005: <timeStamp>
}

If we store it this way, we can direct access the recipe by iD and it may save us from having to iterate through the array when adding / removing recipes.

So I think we can use the Schema.Types.Mixed type so we can store objects.
```suggestion
    recipes: { type: Schema.Types.Mixed, required: true, default: {} }
```